projects/01-rubygems-cli/solution/lib/cache/db_cache.rb

```diff
@@ -0,0 +1,66 @@
+require "sqlite3"
+require "time"
+require "fileutils"
+
+require "./lib/cache/cache_entry"
+
+class DBCache
+  DB_PATH = "./cache.sqlite3"
+  TABLE_NAME = "RUBYGEMS"
+  EXPIRATION_TIME = 48 * 60 * 60
+
+  def initialize
+    @db = SQLite3::Database.new(DB_PATH)
+    @db.execute <<-SQL
+      CREATE TABLE IF NOT EXISTS #{TABLE_NAME} (
+        key VARCHAR(16) PRIMARY KEY,
+        value TEXT,
+        expires_at DATETIME
+      );
+    SQL
+  end
+
+  private_class_method :new
+
+  def read(key)
+    rows = @db.execute <<-SQL
```

Comment:
You'll probably enjoy ActiveRecord once we got to Rails. You can do stuff like this query in a single line.